From 2e52c71caa0e5bbd6e0d618728e4f09b271c6dfb Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Tue, 22 Sep 2026 20:57:46 +0200
Subject: [PATCH] avformat/http: trim trailing whitespace from cookie attribute
 names

RFC 6265 section 5.2 trims whitespace around the attribute name before
it is matched
---
 libavformat/http.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libavformat/http.c b/libavformat/http.c
index 81b3e3489893..dac46d299131 100644
--- a/libavformat/http.c
+++ b/libavformat/http.c
@@ -1092,6 +1092,9 @@ static int parse_set_cookie(const char *set_cookie, AVDictionary **dict)
         next_param = NULL;
         param += strspn(param, WHITESPACES);
         if ((name = av_strtok(param, "=", &value))) {
+            char *end = name + strlen(name);
+            while (end > name && strchr(WHITESPACES, end[-1]))
+                *--end = '\0';
             if (av_dict_set(dict, name, value ? value : "", 0) < 0) {
                 av_free(cstr);
                 return -1;
